com.wamas.ide.launching/src/com/wamas/ide/launching/generator/BundleFilter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) SSI Schaefer IT Solutions
+ */
+package com.wamas.ide.launching.generator;
+
+import java.util.Collection;
+import java.util.function.Predicate;
+
+import org.eclipse.osgi.service.resolver.BundleDescription;
+
+public interface BundleFilter extends Predicate<BundleDescription> {
+
+    public static class Factory {
+
+        public static BundleFilter getBundleIsResolvedFilter() {
+            return BundleDescription::isResolved;
+        }
+
+        public static BundleFilter getBundleIsNotIgnoredFilter(Collection<BundleDescription> toIgnore) {
+            return (b) -> !toIgnore.contains(b);
+        }
+
+        public static BundleFilter and(Iterable<? extends Predicate<BundleDescription>> predicates) {
```

Comment:
better use java.util.function.Predicate.and(Predicate<? super T>)

Reply:
Yeah - failed to find the proper Xtend syntax in the first attempt. Now I can drop BundleFilter again.